### PR TITLE
Move JS interface definition to a subdirectory

### PR DESCRIPTION
**Pull request by magik6k**
This is in preparation for https://github.com/ipfs/interface-ipfs-core/issues/66 / https://github.com/ipfs/go-ipfs/issues/4498.

* Why not `{src,test}/{js,go}`
  * Go usually has the test files in one directory together with the sources
  * Go import path would look weird: `github.com/ipfs/interface-ipfs-core/src/go/`.. vs `github.com/ipfs/interface-ipfs-core/go/`

### CODE REVIEW

**Comment by diasdavid** [comment]
@magik6k sounds good to me. Can you also update the README?

**Comment by magik6k** [comment]
@diasdavid done (assuming that's what you meant)

File: README.md [hunk 1]
```diff
@@ -102,6 +102,11 @@ In order to be considered "valid", an IPFS core implementation  must expose the
   - [Miscellaneous](/SPEC/MISCELLANEOUS.md)
   - [config](/SPEC/CONFIG.md)
 
+Language-specific interface definitions and tests are in their directories:
```
Comment (diasdavid) [inline README.md]: You only find the tests on the `js/src` folder. The interface definitions remain at `/SPEC`

**Comment by diasdavid** [comment]
Can we have at least one of the go-ipfs core APIs spec'ed and tested out just to see how it feels?

**Comment by magik6k** [comment]
> Can we have at least one of the go-ipfs core APIs spec'ed and tested out just to see how it feels?

For the code the plan is to just move https://github.com/ipfs/go-ipfs/tree/master/core/coreapi/interface here into `go/`. Written specs will probably just be docs from the interface code C-c C-v'd into /SPECS docs.

I removed the readme change as it didn't really make sense.

**Comment by lgierth** [comment]
I'm not sure how well gx deals with packages in repo subdirectories. If it turns out that we have to perform in-depth surgery on gx, I'd recommend putting the Go interfaces into their own repo.

Could you try that out before we merge this PR?

**Comment by magik6k** [comment]
So I did `gx init` in `$GOPATH/github.com/ipfs/interface-ipfs-core/go`, it worked fine without any hacks, though only go/ directory was pushed to gx. This is go/package.json I got:
```json
{
  [...]
  "bugs": {
    "url": "https://github.com/ipfs/interface-ipfs-core"
  },
  "gx": {
    "dvcsimport": "github.com/ipfs/interface-ipfs-core/go"
  },
  "gxVersion": "0.11.0",
  [...]
  "name": "go",
}
```

Go tooling didn't have any problems with importing this package (though if a user wants to go-get it it needs to be `github.com/ipfs/interface-ipfs-core/go` (note the `/go`)).

I'd say that it works well enough to have these things in one repo

**Comment by diasdavid** [comment]
> Go tooling didn't have any problems with importing this package (though if a user wants to go-get it it needs to be github.com/ipfs/interface-ipfs-core/go (note the /go)).

That's excellent! Thank you @magik6k ❤️

Mind rebasing master onto this PR to fix merge conflicts?

**Comment by magik6k** [comment]
@diasdavid done

**Pull request merged**